pkg/monitor/cluster/validatepermissions.go

```diff
@@ -0,0 +1,46 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/ARO-RP/pkg/validate/dynamic"
+)
+
+/***************************************************************
+	Monitor the Cluster Service Prinicpal required permissions:
+	- Network Contributor role on vnet
+****************************************************************/
+
+func (mon *Monitor) emitValidatePermissions(ctx context.Context) error {
```

Comment:
I would also check against nil on the pointers that can be nil (like mon.oc, mon.validator...) and return an error if they are nil (I have added that to our Slack conversation).

Reply:
done